contracts/clients/README.md

```diff
@@ -157,34 +155,65 @@ const wallet = await BlsWalletWrapper.connect(
   provider,
 );
 const aggregator = new Aggregator("https://arbitrum-goerli.blswallet.org");
-const estimateFee = await aggregator.estimateFee(bundle); // Remember to include no payment with this bundle
+
+// Create a fee estimate bundle
+const estimateFeeBundle = wallet.sign({
+  nonce,
+  actions: [
+    ...actions, // ... add your user actions here (approve, transfer, etc.)
+    {
+      ethValue: 1,
+      // Provide 1 wei with this action so that the fee transfer to
+      // tx.origin can be included in the gas estimate.
+      contractAddress: aggregatorUtilitiesContract.address,
+      encodedFunction:
+        aggregatorUtilitiesContract.interface.encodeFunctionData(
+          "sendEthToTxOrigin",
+        ),
+    },
+  ],
+});
+
+const feeEstimate = await aggregator.estimateFee(estimateFeeBundle);
+
+// Add a safety premium to the fee to account for fluctuations in gas estimation
+const safetyDivisor = 5;
+const feeRequired = BigNumber.from(feeEstimate.feeRequired);
+const safetyPremium = feeRequired.div(safetyDivisor);
+const safeFee = feeRequired.add(safetyPremium);
 
 const bundle = wallet.sign({
-    nonce: await wallet.Nonce(),
-    actions: [
-        ...actions, // ... do your transaction/actions here (approve, transfer, etc.)
-
-        // then, if the aggregator is using native chain currency, such as ETH
-        {
-            ethValue: estimateFee.feeRequired, // fee amount
-            contractAddress: aggregatorAddress,
-            encodedFunction: "0x"
-        }
-        // or you can use the AggregatorUtilities.sol contract
-        // this makes the bundle submittable from any ETH paid aggregator
-        {
-            ethValue: estimateFee.feeRequired, // fee amount
-            contractAddress: aggregatorUtilitiesContract.address,
-            encodedFunction: aggregatorUtilitiesContract.interface.encodeFunctionData(
-                'sendEthToTxOrigin', [],
-            ),
-        }
-    ],
+  nonce: await wallet.Nonce(),
+  actions: [
+    ...actions, // ... add your user actions here (approve, transfer, etc.)
+    {
+      ethValue: safeFee, // fee amount
+      contractAddress: aggregatorUtilitiesContract.address,
+      encodedFunction:
+        aggregatorUtilitiesContract.interface.encodeFunctionData(
+          "sendEthToTxOrigin",
+        ),
+    },
+
+    // You can also just add an action that transfers the fee amount to the aggregator address, instead of
```

Comment:
Because the aggregator detects it gets paid by observing the effect of a bundle on its own balance, we have different options for paying the aggregator. The first action in this code block is one way of paying the aggregator, and this second action is another way of paying the aggregator, by transferring it funds directly instead of calling `sendEthToTxOrigin`. So the example is supposed to show both ways of doing things.

I've split this into a different code block and updated the description with slightly more context. Let me know if that doesn't make sense still.